app/main.py

```diff
@@ -1,25 +1,83 @@
 from abc import ABC
+from typing import Type, Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self,
+                 min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, owner: Any) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value should be an integer")
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(
+                f"Value should not be less than {self.min_amount} "
+                f"and greater than {self.max_amount} "
+            )
```

Comment:
You should set value
```suggestion
        if not isinstance(value, int):
            raise TypeError("Value should be an integer")
        if not self.min_amount <= value <= self.max_amount:
            raise ValueError(
                f"Value should not be less than {self.min_amount} "
                f"and greater than {self.max_amount} "
            )
        setattr(instance, self.protected_name, value)
```